rust-runtime/aws-smithy-http-server/src/operation/upgrade.rs

```diff
@@ -217,10 +217,9 @@ where
     }
 }
 
-/// Provides an interface to convert a representation of an operation to a HTTP [`Service`](tower::Service) with
-/// canonical associated types.
+/// An interface to convert a representation of a Smithy operation into a [`Route`].
 pub trait Upgradable<Protocol, Operation, Exts, B, Plugin> {
-    /// Performs an upgrade from a representation of an operation to a HTTP [`Service`](tower::Service).
+    /// Upgrade the representation of a Smithy operation to a [`Route`].
```

Comment:
Why these doc changes? I think the user will never care about the `Route` struct per se, but rather what is distinctive about it is that it's an HTTP service, i.e. it `impl<B> Service<Request<B>> for Route<B>`, and so they can feed it to their server.

Reply:
See https://github.com/awslabs/smithy-rs/pull/1954#discussion_r1016539587.
Happy to take them out of this PR, since they are out of scope.